index.html

```diff
@@ -35,83 +46,956 @@
+[NoInterfaceObject, Exposed=Window, CheckAnyPermissions="contextMenus"]
+interface BrowserContextMenusAPI {
+  readonly attribute BrowserContextMenus contextMenus; 
+}
+</pre>
+                    3) Finally, we connect the <code>contextMenus</code> object to the <code>browser.*</code> object.</br>
+<pre class="def idl">
+Browser implements BrowserContextMenusAPI;
+</pre>
+                </section>
+                <section>
+                    <h4>Example (i18n)</h4>
+
+                    In the next example, the <code>i18n</code> API should be made available to both browser host scripts (background, event, popup, options, etc.) and content scripts, and use of this API does not require a manifest declaration. Additionally, however, the manifest does specify a value (<code>default_locale</code>) that is necessary, but cannot be specified by the extension author in script code. This is expressed as a <code>BrowserExtensionManifest</code> dictionary.<br />
+                    <br />
+                    1) First, declare the methods for the <code>browser.contextMenus</code> object. Other methods would follow <code>getMessage</code>, but they are omitted in this example for simplicity.<br />
+<pre class="def idl">
+[NoInterfaceObject]
+interface BrowserI18n {
+  DOMString getMessage(DOMString messageName, sequence<DOMString>? substitutions);
+  void otherMethodsOmittedInThisExampleForSimplicity();
+};
+</pre>
+
+                    2) Next, we specify the available contexts ("Window" for background/event/popup/options...) and "ContentScript" for the content context.
+<pre class="def idl">
+[NoInterfaceObject, Exposed=(Window,ContentScript)] interface BrowserI18nAPI {
+  readonly attribute BrowserI18n i18n;
+}
+</pre>
+                    3) Connect the <code>i18n</code> object to the <code>browser.*</code> object.
+<pre class="def idl">
+Browser implements BrowserI18nAPI;
+</pre>
+                    4) Finally, we express the locale from manifest.json. For example <code>"default_locale": "en"</code>
+<pre class="def idl">
+partial dictionary BrowserExtensionManifest {
+  DOMString? defaultLocale;
+};
+</pre>
+                </section>
+            </section>
+
+        </section>
+
+        <section>
+            <font style="background:yellow;">DISCUSS: It has been suggested that we cut this section.</font><br />
+            <h3>Events</h3>
+            Extension events are similar to DOM Events [<a href="https://www.w3.org/TR/dom/">DOM4</a>]. However, extension events are dispatched only to objects in the relevant extension scope (background, event, options, content, etc.). Script in the extension scope can observe events by calling
+            <code>addListener()</code>.<br />
+            <font style="background:yellow;">TODO: Need to define callbacks throughout this spec, rather than generic "Function callback".</font><br />
+<pre class="def idl">
+[Exposed=(Window,ContentScript) CheckAnyPermissions="browserextensions"]
```

Comment:
This should probably be `NoInterfaceObject` rather than `Exposed` and `CheckAnyPermissions`. And there need to be commas between the attributes.